src/v/cloud_storage/remote_segment.cc

```diff
@@ -616,7 +616,11 @@ ss::future<> remote_segment::run_hydrate_bg() {
             }
         }
     } catch (const ss::broken_condition_variable&) {
-        vlog(_ctxlog.debug, "Hydraton loop is stopped");
+        vlog(_ctxlog.debug, "Hydration loop shut down");
```

Comment:
Shutting down connections early sounds good to me. If we're worried about the extra error states we might enter now that we shut down connections early or if there's any concern about hiding legitimate issues with these catches, I wonder if we could plumb the `remote` abort source into the `remote_segment`s.

Reply:
I would like that in a followup, currently remote_segment has a gate but no abort source, which makes things weird.

Relatedly in PR https://github.com/redpanda-data/redpanda/pull/7193 there are places I want to use sleep_abortable but can't because there's no abort source.